src/formatDate.js

```diff
@@ -8,7 +8,84 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatObject = arrayInObject(fromFormat);
+  const toFormatObject = arrayInObject(toFormat);
+  const arrayOldDate = date.split(fromFormat[3]);
+  const dateObject = arrayDateFromObject(arrayOldDate, fromFormat);
+
+  if (fromFormatObject.yy !== toFormatObject.yy) {
+    if (fromFormatObject.yy.length === 4) {
+      dateObject.yy = dateObject.yy.slice(2);
+    } else {
+      if (+dateObject.yy < 30) {
+        dateObject.yy = '20' + dateObject.yy;
+      } else {
+        dateObject.yy = '19' + dateObject.yy;
+      }
+    }
+  }
+
+  return resultDate(dateObject, toFormat);
+}
+
+function resultDate(dateObject, toFormat) {
+  const result = [];
+
+  for (let i = 0; i < 4; i++) {
+    if (toFormat[i] === 'YYYY' || toFormat[i] === 'YY') {
+      result.push(dateObject.yy);
+    }
+
+    if (toFormat[i] === 'MM') {
+      result.push(dateObject.mm);
+    }
+
+    if (toFormat[i] === 'DD') {
+      result.push(dateObject.dd);
+    }
+  }
```

Comment:
The loop in the `resultDate` function is hardcoding the number of iterations to 4, which assumes the format will always have 3 date parts and a separator. This may not be flexible if the format changes. Consider iterating based on the length of `toFormat` minus one (to exclude the separator).